homeassistant/core.py

```diff
@@ -393,12 +393,28 @@ async def async_block_till_done(self) -> None:
         """Block until all pending work is done."""
         # To flush out any call_soon_threadsafe
         await asyncio.sleep(0)
+        start_time = None  # type: Optional[float]
```

Comment:
```suggestion
        start_time: Optional[float] = None
```

Reply:
Yes, that's much better.  👍   Thats what I get for working on this at the end of the day.